File: leetcode/101-200/190-reverse-bits/bench/reverse_bits.py
"""Benchmark workload for LeetCode #190 — Reverse Bits (Python; scale lane)."""


def reverse_bits(n):
    result = 0
    x = n
    for _ in range(32):
        result = (result << 1) | (x & 1)
        x >>= 1
    return result


def main():
    count = 8000000
    state = 12345
    sink = 0
    for _ in range(count):
        state = (state * 1103515245 + 12345) & 2147483647
        sink += reverse_bits(state)
    print(sink)


main()

File: leetcode/101-200/190-reverse-bits/bench/reverse_bits.c
#include <stdio.h>

static long reverse_bits(long n) {
    long result = 0;
    long x = n;
    for (long i = 0; i < 32; i++) {
        result = (result << 1) | (x & 1);
        x >>= 1;
    }
    return result;
}

int main(void) {
    long count = 8000000;
    long state = 12345;
    long sink = 0;
    for (long i = 0; i < count; i++) {
        state = (state * 1103515245 + 12345) & 2147483647;
        sink += reverse_bits(state);
    }
    printf("%ld\n", sink);
    return 0;
}
